calculator.rb

```diff
@@ -0,0 +1,154 @@
+# Calculator Exercise
+# Build a calculator command line interface (CLI) that allows a user to perform simple arithmetic.
+
+# Welcome message:
+puts "\n"
+puts "Welcome to this super basic calculator!"
+puts "\n"
+
+
+
+# Prompt the user for a operation to be made on the calculator:
+print "What operation would you like to do? "
+operation = gets.chomp
+valid_operations = %w[add + addition subtract - subtraction multiply *
+  multiplication divide division / modulo % exponential ^]
+  while !(valid_operations.include?(operation))
+    puts "Input invalid."
+    print "Please enter a basic operation to be made: "
+    operation = gets.chomp
+  end
+
+
+  # Prompt the user for the numbers to be used on the calculation:
+  puts "\n"
+  puts "What are the numbers you would like to perform the operation with?"
+
+  puts "\n"
+  print "Number 1: "
+  first_num = gets.chomp
+  while first_num.match?(/\D/) && !(first_num.match?(/\d\.\d/)) || first_num == ""
```

Comment:
Nice work trapping user input until they get the put in an acceptable value.  

However notice that you repeat this code... makes a good candidate for a method!